### ISSUE TITLE
Replace Conv to MPL

### ISSUE THREAD

**Issue by zhongming12342** [thuml/Transolver#18]
Hi! Thank you for your work!!

In you core code, “Physics_Attention_Structured_Mesh_2D” class in PDE-Solving-StandardBenchmark/model/Transolver_Structured_Mesh_2D.py, you use the following forward function:
      
        x = x.reshape(B, self.H, self.W, C).contiguous().permute(0, 3, 1, 2).contiguous()  # B C H W

        ### (1) Slice
         fx_mid = self.in_project_fx(x).permute(0, 2, 3, 1).contiguous().reshape(B, N, self.heads, self.dim_head) \
             .permute(0, 2, 1, 3).contiguous()  # B H N C
         x_mid = self.in_project_x(x).permute(0, 2, 3, 1).contiguous().reshape(B, N, self.heads, self.dim_head) \
             .permute(0, 2, 1, 3).contiguous()  # B H N G
        
         slice_weights = self.softmax(
             self.in_project_slice(x_mid) / torch.clamp(self.temperature, min=0.1, max=5))  # B H N G
         slice_norm = slice_weights.sum(2) 
         slice_token = torch.einsum("bhnc,bhng->bhgc", fx_mid, slice_weights)
         slice_token = slice_token / ((slice_norm + 1e-5)[:, :, :, None].repeat(1, 1, 1, self.dim_head))

        ### (1) Get qkv
         q_slice_token = self.to_q(slice_token)
         k_slice_token = self.to_k(slice_token)
         v_slice_token = self.to_v(slice_token)

        ### (2) Attention among slice tokens

         dots = torch.matmul(q_slice_token, k_slice_token.transpose(-1, -2)) * self.scale
         attn = self.softmax(dots)
         attn = self.dropout(attn)
         out_slice_token = torch.matmul(attn, v_slice_token)  # B H G D

         ### (3) Deslice
         out_x = torch.einsum("bhgc,bhng->bhnc", out_slice_token, slice_weights)
         out_x = rearrange(out_x, 'b h n d -> b n (h d)')    

which seems strange. When i chaned them to MLP as follows:
###
        slice_weights = self.softmax(
            self.in_project_slice(x) / torch.clamp(self.temperature, min=0.1, max=5))  
        slice_norm = slice_weights.sum(1)  # B  G
        slice_token = torch.einsum("bnc,bnm->bmc", x, slice_weights)
        slice_token = slice_token / ((slice_norm[...,None] + 1e-5))
        q_slice_token = self.to_q(slice_token).reshape(B,self.slice_num, self.heads, self.dim_head).permute(0, 2, 1, 3)  
        k_slice_token = self.to_k(slice_token).reshape(B,self.slice_num, self.heads, self.dim_head).permute(0, 2, 1, 3)
        v_slice_token = self.to_v(slice_token).reshape(B,self.slice_num, self.heads, self.dim_head).permute(0, 2, 1, 3)

        dots = torch.matmul(q_slice_token, k_slice_token.transpose(-1, -2)) * self.scale
        attn = self.softmax(dots)
        attn = self.dropout(attn)
        out_slice_token = rearrange(torch.matmul(attn, v_slice_token),'b h m d -> b m (h d)')  # B M D

        ### (3) Deslice
        out_x = torch.einsum("bmc,bnm->bnc", out_slice_token, slice_weights)
